src/ser.rs

```diff
@@ -90,30 +98,50 @@ impl<'py> Serialize for SerializePyObject<'py> {
             for (k, v) in simple_items {
                 let key = table_key(k)?;
                 let value = self.with_obj(v);
+                let v_ob_type = v.get_type_ptr() as usize;
+                if v_ob_type == lookup.none && self.omit_none {
+                    continue;
+                }
                 map.serialize_entry(key, &value)?;
             }
             for (k, v) in array_items {
                 let key = table_key(k)?;
                 let value = self.with_obj(v);
+                let v_ob_type = v.get_type_ptr() as usize;
+                if v_ob_type == lookup.none && self.omit_none {
+                    continue;
+                }
                 map.serialize_entry(key, &value)?;
             }
             for (k, v) in dict_items {
                 let key = table_key(k)?;
                 let value = self.with_obj(v);
+                let v_ob_type = v.get_type_ptr() as usize;
+                if v_ob_type == lookup.none && self.omit_none {
+                    continue;
+                }
                 map.serialize_entry(key, &value)?;
             }
             map.end()
         } else if ob_type == lookup.list {
             let py_list: &PyList = self.obj.cast_as().map_err(map_py_err)?;
             let mut seq = serializer.serialize_seq(Some(py_list.len()))?;
             for element in py_list {
+                let v_ob_type = element.get_type_ptr() as usize;
+                if v_ob_type == lookup.none && self.omit_none {
+                    continue;
```

Comment:
same as tuple case.

Reply:
Sorry, ignore my previous comment.

What confuses me is that we have L126 to handle list (array) items and here we have L106 to handle the list again. I suppose that L106 is handling a list nested deeply in a dict.

Should we consider the None in the nested list here?

Reply:
It's for the order of items, the comment got removed in the the recent updates

```
// https://github.com/alexcrichton/toml-rs/issues/142#issuecomment-278970591
// taken from alexcrichton/toml-rs/blob/ec4e821f3bb081391801e4c00aa90bf66a53562c/src/value.rs#L364-L387
```